ext/bg/js/search.js

```diff
@@ -241,39 +239,63 @@ class DisplaySearch extends Display {
     initClipboardMonitor() {
         // ignore copy from search page
         window.addEventListener('copy', () => {
-            this.clipboardPrevText = document.getSelection().toString().trim();
+            this.clipboardPreviousText = document.getSelection().toString().trim();
         });
     }
 
     startClipboardMonitor() {
-        this.clipboardMonitorIntervalId = setInterval(async () => {
-            let curText = null;
-            // TODO get rid of this and figure out why apiClipboardGet doesn't work on Firefox
-            if (IS_FIREFOX) {
-                curText = (await navigator.clipboard.readText()).trim();
-            } else if (IS_FIREFOX === false) {
-                curText = (await apiClipboardGet()).trim();
-            }
-            if (curText && (curText !== this.clipboardPrevText) && jpIsJapaneseText(curText)) {
-                if (this.isWanakanaEnabled()) {
-                    this.setQuery(window.wanakana.toKana(curText));
-                } else {
-                    this.setQuery(curText);
+        const token = {};
```

Comment:
Is this used to avoid overlapping callbacks? First time I see an object used like this. :smile: Maybe there could be a comment explaining why the object isn't used to store data.

Reply:
Yeah, it's effectively a unique identifier used to indicate that the current callback is still active. An incrementing counter could could be similarly used, but technically that has issues with overflow (which in practice would likely not be encountered).

This "token" is comparing to see that the reference is the same, which is only true for the same object. If it's different, a new timer has been started during the `await`.